AutoRest/Generators/Extensions/Extensions/Extensions.cs

```diff
@@ -89,16 +115,33 @@ public static void ProcessParameterizedHost(ServiceClient serviceClient, Setting
                                 parameter.ClientProperty = serviceClient.Properties.Single(p => p.SerializedName.Equals(parameter.SerializedName));
                             }
                             parameter.Extensions["hostParameter"] = true;
+                            hostParamList.Add(parameter);
+                        }
 
-                            foreach (var method in serviceClient.Methods)
+                        foreach (var method in serviceClient.Methods)
+                        {
+                            if (position.Equals("first", StringComparison.OrdinalIgnoreCase))
                             {
-                                method.Parameters.Add(parameter);
+                                method.Parameters.InsertRange(0, hostParamList);
```

Comment:
@NiklasGustafsson This step happens in the extensions which is before the parameters are sorted into two buckets (required, optional). We preserve the order within these two buckets. So, if the positionInOperation is first, then all the parameters in the host template that will be local parameters in the method, will be added in the beginning of all the required parameters and then we will have all the optional parameters. The x-ms-parameterized-host extension can only have path parameters.
If the position is last then all the required local parameters in the host template will be appended to the list of required parameters for a particular method. All the optional parameters will appear after that.

We can definitely make "first" as the default position. The user can explicitly chose to add those parameters in the end.